src/components/NewRoomModal.tsx

```diff
@@ -0,0 +1,167 @@
+'use client'
+import { useState, useRef, useCallback, useEffect } from 'react'
+import { useRouter } from 'next/navigation'
+
+// import { socket } from '@/socket'
+
+import RightArrowIcon from './icons/RightArrowIcon'
+
+export default function NewRoomModal() {
+	const [isLoading, setIsLoading] = useState(true)
+	const [roomUrl, setRoomUrl] = useState('')
+	const [hostRoomUrl, setHostRoomUrl] = useState('')
+	const [hideGoBtn, setHideGoBtn] = useState(true)
+	const [hideCreateBtn, setHideCreateBtn] = useState(true)
+	const [showNameInput, setShowNameInput] = useState(true)
+	const [domain, setDomain] = useState('')
+	const roomUrlRef = useRef<HTMLSpanElement>(null)
+	const dialogRef = useRef<HTMLDialogElement>(null)
+	const hostRef = useRef<HTMLInputElement>(null)
+	const [hostFirstName, setHostFirstName] = useState('')
+
+	const router = useRouter()
+
+	useEffect(() => {
+		if (typeof window !== 'undefined') {
+			const hostname = window.location.hostname
+			const http = hostname === 'localhost' ? 'http' : 'https'
+			const port = window.location.port ? `:${window.location.port}` : ''
+			setDomain(`${http}://${hostname}${port}`)
+		}
+		const hostDataLocalStorage = localStorage.getItem('scrumDivingHostData')
+		if (hostDataLocalStorage) {
+			const hostData = JSON.parse(hostDataLocalStorage)
+			if (hostData?.hostName) {
+				setHostFirstName(hostData?.hostName)
+				setHideCreateBtn(false)
+				setShowNameInput(false)
```

Comment:
suggestion/todo: I believe you have way too many states in this component. Each change in the DOM is a new state, but you don't need too. Here's a tip to better manage your states state: Instead of calling states based on its outcome in the DOM (hideGoBtn, hideCreateBtn, etc), try to name it based on the data that we really need to mutate the DOM and then use that data to change the DOM.

For example, in this case the state `hostFirstName` is enough to drive the DOM mutation `hideCreateBtn`, so you don't need that extra state. Instead of `disabled={hideCreateBtn}`, you can say `disabled={!!hostFirstName}`.

Does this make sense to you?

Reply:
I've never done that, converted a string to a boolean with `!!`. But I get it. I'll work on that.

Reply:
This has been entirely reworked. I'll push in the next commit.

Reply:
DONE